src/transformStateWithClones.js

```diff
@@ -6,8 +6,34 @@
  *
  * @return {Object[]}
  */
+
 function transformStateWithClones(state, actions) {
-  // write code here
+  let newState = { ...state };
+  const output = [];
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
```

Comment:
Consider using a switch case here

Reply:
I got so used to if / else if that I completely forgot about switch. Thank you